### PR TITLE
Cli configurable validators

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

`solana validators` can only return a filtered list of validators

#### Summary of Changes

- add config parameters for `getVoteAccounts`
- plumb them through CLI

### CODE REVIEW

File: rpc/src/rpc.rs [hunk 1]
```diff
@@ -837,22 +840,27 @@ impl JsonRpcRequestProcessor {
...
                 } else {
                     vote_account_info.last_vote > 0
                 }
             });
 
-        let delinquent_staked_vote_accounts = delinquent_vote_accounts
-            .into_iter()
-            .filter(|vote_account_info| vote_account_info.activated_stake > 0)
-            .collect::<Vec<_>>();
+        let keep_unstaked_delinquents = config.keep_unstaked_delinquents.unwrap_or(false);
```
Comment (CriesofCarrots) [inline rpc/src/rpc.rs:851]:
```suggestion
        let keep_unstaked_delinquents = config.keep_unstaked_delinquents.unwrap_or_default();
```

File: cli/src/cluster_query.rs [hunk 2]
```diff
@@ -382,6 +383,25 @@ impl ClusterQuerySubCommands for App<'_, '_> {
...
+                        .help("Don't discard unstaked, delinquent validators")
+                )
+                .arg(
+                    Arg::with_name("delinquent_slot_distance")
+                        .long("delinquent-slot-distance")
+                        .takes_value(true)
+                        .value_name("SLOT_DISTANCE")
+                        .validator(is_slot)
+                        .help(
+                            concatcp!(
```
Comment (CriesofCarrots) [inline cli/src/cluster_query.rs:400]: TIL! Darn, could have used this recently

**Review by CriesofCarrots — COMMENTED** [review]
Code lgtm, one nit.
Concerned at all about the implications of variable delinquent slot distance for downstream tools (block explorers, validator stats, etc)? Or you figure that anyone assertive enough to use `delinquent_slot_distance` should expose that configuration to users in their product?

**Comment by t-nelson** [comment]
> Code lgtm, one nit.
> Concerned at all about the implications of variable delinquent slot distance for downstream tools (block explorers, validator stats, etc)? Or you figure that anyone assertive enough to use `delinquent_slot_distance` should expose that configuration to users in their product?

I'd like to think that no one would bother to expose it at all.  This does remind me that I should update the docs though, so I'll put a warning there.

File: cli/src/cluster_query.rs (hunk 2 above)
Comment (t-nelson) [reply cli/src/cluster_query.rs:400]: Ha!  I thought I was going to be clever and use `concat!()` which I learned from your `storage-proto` `build.rs`, but it blew up on the constant.  Crates.io to the rescue!

**Comment by CriesofCarrots** [comment]
> I'd like to think that no one would bother to expose it at all

If "http://validator-website.com" is calling my node delinquent for being 20 slots behind, but my node looks fine on "http://validator-website-2.com" because they are using a threshold of 300 slots, I would want to know?

**Comment by t-nelson** [comment]
> > I'd like to think that no one would bother to expose it at all
> 
> If "http://validator-website.com" is calling my node delinquent for being 20 slots behind, but my node looks fine on "http://validator-website-2.com" because they are using a threshold of 300 slots, I would want to know?

Sorry, I misread that as _expose_ as a knob to the user.  I'd really like to think no one would pass it, period.  The following note of discouragement was added with the docs update.  Hopefully it's sufficient

https://github.com/solana-labs/solana/blob/9ed1f24188cf53d6950abab9a98df8e57b789285/docs/src/developing/clients/jsonrpc-api.md#L3032

**Pull request merged**